packages/sign_in_with_apple/ios/Classes/SwiftSignInWithApplePlugin.swift

```diff
@@ -111,99 +129,161 @@ public class SwiftSignInWithApplePlugin: NSObject, FlutterPlugin {
 }
 
 @available(iOS 13.0, *)
-class AYSignInWithAppleAuthorizationController: NSObject, ASAuthorizationControllerDelegate {
-    enum CurrentAction {
-        case attemptWithPasswords
-        case attemptWithoutPasswords
+class SignInWithAppleAuthorizationController: NSObject, ASAuthorizationControllerDelegate {
+    var callback: FlutterResult
+    
+    var requests: [ASAuthorizationRequest]
+
+    init(_ requests: [ASAuthorizationRequest], callback: @escaping FlutterResult) {
+        self.requests = requests
+        self.callback = callback
     }
-
-    var resultCallback: FlutterResult
-
-    var currentAction: CurrentAction = .attemptWithPasswords
-
-    init(_ callback: @escaping FlutterResult) {
-        resultCallback = callback
+    
+    /// Parses a list of json requests into the proper [ASAuthorizationRequest] type.
+    ///
+    /// The parsing itself tries to be as lenient as possible to recover gracefully from parsing errors.
+    public static func parseRequests(rawRequests: [Any]) -> [ASAuthorizationRequest] {
+        var requests: [ASAuthorizationRequest] = []
+        
+        for request in rawRequests {
+            guard let requestMap = request as? [String: Any] else {
+                print("[SignInWithApplePlugin]: Request is not an object");
+                continue
+            }
+            
+            guard let type = requestMap["type"] as? String else {
+                print("[SignInWithApplePlugin]: Request type is not an string");
+                continue
+            }
+            
+            switch (type) {
+            case "appleid":
+                let appleIDProvider = ASAuthorizationAppleIDProvider()
+                let appleIDRequest = appleIDProvider.createRequest()
+                
+                if let scopes = requestMap["scopes"] as? [String] {
+                    appleIDRequest.requestedScopes = []
+                    
+                    for scope in scopes {
+                        switch scope {
+                        case "email":
+                            appleIDRequest.requestedScopes?.append(.email)
```

Comment:
Force unwrap as we just initialized it above? Or else we could build out own list, and set the final value to the request?

Reply:
Building our own list was a bit more explicit and I removed it.

While we know that this shouldn't be `nil` here, I don't see a big benefit here by force unwrapping, which could lead to a full crash.

Reply:
Certainly don't want to crash, but I think using `?` (which might to nothing) in cases where actions are expected could lead to hard-to-discover bugs.

Here it's so close and direct that we know it'll work, but I would be wary if the places are further apart.